### ISSUE TITLE
failed to run `e2e_test/streaming/now.slt`

### ISSUE THREAD

**Issue by lmatz** [risingwavelabs/risingwave#17439]
### Describe the bug

```
# In madsim test, we cannot spawn process.
skipif madsim
# In parallel test, we cannot get the DB name.
skipif parallel
# TODO: Later if we introduce a new `now()`-like function that returns the time of statement execution,
# we'll be able to directly create MV without `./risedev psql` and so that we can remove these `skipif`.
system ok
./risedev psql -c "
create materialized view mv as
select * from generate_series(
    to_timestamp($(date +%s)) - interval '10 second',
    now(),
    interval '1 second'
);
"

skipif madsim
skipif parallel
statement ok
flush;

skipif madsim
skipif parallel
query I
select count(*) >= 10 from mv;
----
t
```

### Error message/log

```text
Caused by:
2024-06-24 16:16:39 UTC	    query failed: db error: ERROR: Failed to run the query
2024-06-24 16:16:39 UTC	
2024-06-24 16:16:39 UTC	    Caused by these errors (recent errors listed first):
2024-06-24 16:16:39 UTC	      1: Catalog error
2024-06-24 16:16:39 UTC	      2: table or source not found: mv
2024-06-24 16:16:39 UTC	
2024-06-24 16:16:39 UTC	    [SQL] select count(*) >= 10 from mv;
2024-06-24 16:16:39 UTC	    at e2e_test/streaming/now.slt:24
```
```


### To Reproduce

_No response_

### Expected behavior

_No response_

### How did you deploy RisingWave?

_No response_

### The version of RisingWave

_No response_

### Additional context

https://buildkite.com/risingwavelabs/main-cron/builds/2682#

This buildkite test's name is called "end-to-end test (parallel, in-memory) (release)"

Is the test expected to be skipped due to `skipif parallel`?

**Comment by BugenZhao** [comment]
Is this fixed by #17435? @stdrc

**Comment by stdrc** [comment]
> Is this fixed by #17435? @stdrc

Yes, should be.
